passes/design.md

```diff
@@ -0,0 +1,47 @@
+## Nanopass Architecture
+
+The idea is to structure compilation as repeatedly transforming the source
+language into increasingly lower-level, well-defined intermediate languages
```

Comment:
Just a note: Technically we can also go back up via a pass designed to do as such. So given language `b`, which is higher level than language `a`, we could have one pass to lower from `b` to `a`, and another to raise from `a` to `b`.

I don't think we need to document this just yet, at least not until we run into an example of the need in the wild, and then we can speak to the conditions under which we might want to do it.